### PR TITLE
fix panic in dict column merge

### PR DESCRIPTION
**Pull request by PSeitz**
seems not to fix the issue reported in discord #1929, but I think it does fix a possible issue

### CODE REVIEW

**Comment by PSeitz** [comment]
https://github.com/quickwit-oss/tantivy/issues/1929

File: columnar/src/columnar/merge/merge_dict_column.rs [hunk 1]
```diff
@@ -52,20 +52,18 @@ impl<'a> Iterable for RemappedTermOrdinalsValues<'a> {
...
-                        .ords()
-                        .values
-                        .iter()
-                        .map(move |term_ord| segment_ord[term_ord as usize])
-                })
-            });
+        let mut ord = 0;
+        let iter = self.bytes_columns.iter().flat_map(move |byte_column| {
+            let segment_ord = self.term_ord_mapping.get_segment(ord as u32);
+            ord += 1;
```
Comment (fulmicoton) [inline columnar/src/columnar/merge/merge_dict_column.rs]: I don't understand the difference !?

Comment (ChillFish8) [reply columnar/src/columnar/merge/merge_dict_column.rs]: If a column is `None`, it won't increase the ordinal. 

The original would still increase the ordinal regardless of if it was `None` or not IIRC

Comment (fulmicoton) [reply columnar/src/columnar/merge/merge_dict_column.rs]: Thanks! That is not how flat_map works however. It is called on None as well.
I suspect this is the bug you encounterred. I think I fixed it, can you try and reproduce with the last commit?

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**